### ISSUE TITLE
[bug] `no-flush-sync` False Positive with Pino

### ISSUE THREAD

**Issue by joealden** [Rel1cx/eslint-react#1943]
### Describe the bug

If you search for `flushSync` in https://github.com/pinojs/pino/blob/main/docs/api.md - you'll find that there is an API for it. The existing `no-flush-sync` rule doesn't look to be doing any checks on where this was imported from.

### Reproduction

```ts
import { destination } from "pino";

destination().flushSync();
```

**Comment by coltupyours-lang** [comment]
Hey guys! Do you guys torture people for a living as a full time job?

**Comment by changbaebang-29cm** [comment]
I reproduced this on the current `main` branch. The rule currently matches only the callee name, so the Pino example, local `flushSync` functions, and unrelated object methods are all reported, while an aliased `react-dom` import is missed.

I'd like to work on this by following the import-tracking pattern already used by the neighboring `no-hydrate` and `no-render` rules: report only direct or member calls that resolve to a `react-dom` import, and add regression coverage for the Pino example, local names, object methods, and aliased imports.

**Comment by changbaebang** [comment]
I reproduced this on the current `main` branch. The rule currently matches only the callee name, so the Pino example, local `flushSync` functions, and unrelated object methods are all reported, while an aliased `react-dom` import is missed.

I'd like to work on this by following the import-tracking pattern already used by the neighboring `no-hydrate` and `no-render` rules: report only direct or member calls that resolve to a `react-dom` import, and add regression coverage for the Pino example, local names, object methods, and aliased imports.
